middleware/TelemetryHandler.md

```diff
@@ -2,18 +2,38 @@
 
 ## Objectives
 
-Provide a middleware component that attaches metadata to a Graph request in order help improve the SDK team improve the developer experience.
+Provide a mandatory middleware component that attaches metadata to a Graph request in order help the SDK team improve the developer experience.
 
 ## Requirements
 
 - Add a `client-request-id` header with GUID value to request if one is not present.  If the [RequestContext](../middleware/RequestContext.md) has a `ClientRequestId` property set then that value should be used.
-- Add `SdkVersion` request header to each request to identify the language and version of the client SDK library.  The product identifier should be of the format `graph-{lang}`. The `SdkVersion` version can contain multiple values, or appear multiple times in a request. Values should be comma delimited as normal header lists are if the `SdkVersion` contains multiple values. e.g.
 
-    `SdkVersion: MyApp/1.0, spfx/1.0, graph-javascript-1.0`
+- Add `SdkVersion` request header to each request to identify the language and version of the client SDK library. The product identifier should be of the format `graph-{lang}`. The `SdkVersion` version can contain multiple values, or appear multiple times in a request. Values should be comma delimited as normal header lists are if the `SdkVersion` contains multiple values. e.g.
```

Comment:
The SdkVersion MUST be updated with information identifying the use of not only the core libraries, but also the service API libraries.

Reply:
For .Net, we can capture the service models' SDK version in our core library via reflections, but I'm not 100% sure we can do the same for other client libraries unless we update our code generation pipeline to capture this.

For now, I'll update to the spec to  treat this as an optional metadata, and we should only capture it when it's available.